### ISSUE TITLE
SigV2 Message Signer

### ISSUE THREAD

**Issue by KasperJanssens** [aws/aws-sdk-java-v2#1024]
Hello,

I need to consume a service that implements the s3 protocol (not amazon) that only talks v2 signature. Is it still possible to use v2 signing in the v2 sdk and if so, how exactly? I've been looking at the service config and the client configuration but I do not seem to find where to configure the signature version. Is it possible to point to some docs where this is described (or tell me that it is no longer supported)?

**Comment by millems** [comment]
We removed SigV2 support in V2, because it had been deprecated for quite some time. Here's the implementation from an old commit, before it was removed: https://github.com/aws/aws-sdk-java-v2/blob/5c8f22b60998547d3399773fedaa8462c3acfa8c/core/auth/src/main/java/software/amazon/awssdk/auth/signer/QueryStringSigner.java . 

It should theoretically still work, if you wanted to try adding it back into your application. You can set it via:

```Java
clientBuilder.overrideConfiguration(c -> c.putAdvancedOption(SdkAdvancedClientOption.SIGNER, 
                                                             QueryStringSigner.create()));
```

Let me get back to you if this is something we want to resurrect in the SDK.

**Comment by KasperJanssens** [comment]
Alright, did something along those lines but implemented my own V2 signer (discovered that Signer option before I saw this answer). I'll check out the string signer and it is good to know that this is indeed the way to go for v2. Thanks. Should I close the ticket yet while I await your answer concerning resurrecting this? Or you prefer keeping it open?

**Comment by millems** [comment]
Let's leave it open. People could want a V2 signer to be shipped with the SDK, so it would be nice to have some stance on this.

**Comment by spullara** [comment]
How do you call services that require v2 signatures? Both importexport and sdb still require it according to their botocore specifications. Is that not accurate?

**Comment by millems** [comment]
`simpledb` and `importexport` are not currently supported in 2.x of the SDK.

**Comment by ndtuan412** [comment]
+1 on this feature for s3 service.
I am consuming a s3 compatible service that only support signature v2.

Tried class `QueryStringSigner` above but it does not work.
Code
```java
System.setProperty("aws.accessKeyId", accessKeyId);
System.setProperty("aws.secretAccessKey", secretKeyId);

S3Client s3 = S3Client.builder()
        .region(Region.AP_SOUTHEAST_1)
        .endpointOverride(new URI("http://xxxx"))
        .credentialsProvider(SystemPropertyCredentialsProvider.create())
        .serviceConfiguration(S3Configuration.builder().pathStyleAccessEnabled(true).build())
        .overrideConfiguration(c -> c.putAdvancedOption(SdkAdvancedClientOption.SIGNER, QueryStringSigner.create()))
        .build();
```
Error
```
Caused by: software.amazon.awssdk.services.s3.model.S3Exception: Access Denied (Service: S3, Status Code: 403, Request ID: ac1044c2:16b07363dee:18f4f:2116f)
```
p/s: I tried other SDK and everything work
- Python: `boto3==1.9.173`
- NodeJs: `"aws-sdk": "^2.390.0"`

I also tried the minio SDK, but the error was `Caused by: io.minio.errors.ErrorResponseException: The request signature we calculated does not match the signature you provided. Check your Secret Access Key and signing method. For more information, see REST Authentication and SOAP Authentication for details.`. Strangely.
MinIO code:
```java
MinioClient minioClient = new MinioClient("xxxxx", 80, accessKeyId, secretKeyId, "ap-southeast-1", false);

List<Bucket> bucketList = minioClient.listBuckets();
for (Bucket bucket : bucketList) {
    System.out.println(bucket.creationDate() + ", " + bucket.name());
}
```

**Comment by millems** [comment]
Unfortunately I don't think this feature is something we want to support in the SDK, because there are no public AWS services that we intend to support that also use SigV2. Signing is pluggable, so it is possible to inject your own implementation if you are completely blocked on this issue.
